hazelcast/src/main/java/com/hazelcast/nio/tcp/handlermigration/IOSelectorLoadCalculator.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.nio.tcp.handlermigration;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.LoggingService;
+import com.hazelcast.nio.tcp.AbstractIOSelector;
+import com.hazelcast.nio.tcp.IOSelector;
+import com.hazelcast.nio.tcp.MigratableHandler;
+import com.hazelcast.nio.tcp.SelectionHandler;
+import com.hazelcast.util.ItemCounter;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import static com.hazelcast.util.StringUtil.getLineSeperator;
+
+/**
+ * Calculates a state describing utilization of IOSelector(s) and creates a mapping between IOSelector -> Handler.
+ *
+ * This class is not thread-safe with the exception of
+ * {@link #addHandler(T)}   and
+ * {@link #removeHandler(T)}
+ *
+ */
+class IOSelectorLoadCalculator<T extends MigratableHandler & SelectionHandler> {
```

Comment:
Instead of generic type `and (&)` operation, I would prefer new interface such as `MigratableSelectionHandler` interface that extends from both of `MigratableHandler` and `SelectionHandler` interfaces